sdks/go/pkg/beam/core/metrics/metrics.go

```diff
@@ -448,3 +453,50 @@ func (m *gauge) get() (int64, time.Time) {
 	defer m.mu.Unlock()
 	return m.v, m.t
 }
+
+// GaugeValue is the value of a Gauge metric.
+type GaugeValue struct {
+	Value     int64
+	Timestamp time.Time
+}
+
+// MetricResults queries for all metric values that match a given filter.
+type MetricResults interface {
+	// TODO: Implement metrics filtering
```

Comment:
Please file a JIRA (https://issues.apache.org/jira/projects/BEAM/issues) under the sdk-go component for this PRs and followup work. Then add the JIRA [BEAM-###] as a prefix to the PR title.